### ISSUE TITLE
Remove dogstatsd exporter

### ISSUE THREAD

**Issue by jmacd** [open-telemetry/opentelemetry-go#543]
This code should move to a contrib or vendor repository.

I had pursued the idea of putting together a specification for statsd + labels like dogstatsd, but I couldn't find much support for this idea. Dogstatsd, while it is widely used outside of Datadog, does not have a specification that we can use.

**Comment by lizthegrey** [comment]
where does this put things like Telegraf that are officially maintained by InfluxDB but in reality are used by everyone?

**Comment by jmacd** [comment]
My feeling is that statsd+labels is a very useful protocol and we should support it.

I've engaged in private conversations, trying to find anyone that will produce a viable statsd+labels protocol spec. Even "plain" statsd lacks a proper specification, and I would support a plain statsd exporter too, although I think it it less useful in practice--and it's wasteful to encode labels into the metric name and parse them out later which is how many systems treat plain statsd (e.g., https://github.com/prometheus/statsd_exporter, https://www.datadoghq.com/blog/dogstatsd-mapper).

I really expected that someone with a vested interested in OTel+STatsd being an official thing would step up here, but since it's so close to our beta, I'm interested in removing non-standard code. This is not my fight--I was trying to help by creating the OTel-Go statsd code. If we did have a spec for statsd+labels, I am opposed to using what Datadog has given us. There is an unsolved issue that to me makes-or-breaks this protocol, https://github.com/DataDog/datadog-go/issues/83, and currently it's broken. What we need is an improved statsd+labels protocol, IMO it's very simple to specify but it should give us details about the character set and support escaping values.
